test/cuda/test_tc_mapper.cc

```diff
@@ -297,6 +297,67 @@ def tensordot(float(N, C1, C2, H, W) I0, float(N, C2, C3, H, W) I1) -> (O) {
   ::benchmarkKernelOptions(TC, name, inputs, options);
 }
 
+///////////////////////////////////////////////////////////////////////////////
+// TensorAdd2D -- Version 0
+//   O(n, m) +=! I0(n, m) * I1(n, m)
+///////////////////////////////////////////////////////////////////////////////
+TEST_F(TcCudaMapperTest, TensorAdd2Dv0) {
+  N = 64; M=64;
+  at::Tensor I0 = at::CUDA(at::kFloat).rand({N,M});
+  at::Tensor I1 = at::CUDA(at::kFloat).rand({N,M});
+  //std::vector<at::Tensor> inputs = {I0, I1};
+  std::vector<at::Tensor> inputs = {I0, I1};
+
+  static constexpr auto TC = R"TC(
+def tensoradd2dv0(float(N, M) I0, float(N, M) I1) -> (O) {
+    O(n, m) +=! I0(n, m) + I1(n, m)
+}
+  )TC";
+  auto checkFun = [](const std::vector<at::Tensor>& ins,
+		  std::vector<at::Tensor>& outs) { return true;};
+  auto options = tc::CudaMappingOptions::makeNaiveMappingOptions();
+  auto name = "tensoradd2dv0";
+  auto res = Check(TC, name, options, inputs, checkFun);
+  std::string expected = "for (int c2 = t1; c2 <= 31; c2 += 8)";
+  ASSERT_NE(std::string::npos, res.second.find(expected))
+          << "In resulting code:\n"
+          << res.second << "\nfound unexpected: " << expected;
+  ::benchmarkKernelOptions(TC, name, inputs, options);
+}
+
+///////////////////////////////////////////////////////////////////////////////
+// TensorAdd2D  -- Version 1 with Strides
+//   O(n, m) +=! I0(n, m) * I1(n, m)
+///////////////////////////////////////////////////////////////////////////////
+TEST_F(TcCudaMapperTest, TensorAdd2Dv1) {
+  N = 64; M=64;
+  at::Tensor I0 = at::CUDA(at::kFloat).rand({N,M});
+  at::Tensor I0_view = I0.type().tensor().set_(*I0.storage(), 0 /*offset*/,{N,M}/* size*/, {1,16}/*strides*/);
+  at::Tensor I1 = at::CUDA(at::kFloat).rand({N,M});
+  at::Tensor I1_view = I1.type().tensor().set_(*I1.storage(), 0 /*offset*/,{N,M}/* size*/, {1,16}/*strides*/);
+  //std::vector<at::Tensor> inputs = {I0, I1};
+  std::vector<at::Tensor> inputs = {I0_view, I1_view};
+
+  static constexpr auto TC = R"TC(
+def tensoradd2dv1(float(N, M) I0_view, float(N, M) I1_view) -> (O) {
+    O(n, m) +=! I0_view(n, m) + I1_view(n, m)
+}
+  )TC";  
+
+  auto checkFun = [](const std::vector<at::Tensor>& ins,
+		  std::vector<at::Tensor>& outs) { return true;};
+  auto options = tc::CudaMappingOptions::makeNaiveMappingOptions();
+  auto name = "tensoradd2dv1";
+  auto res = Check(TC, name, options, inputs, checkFun);
+  //This test should be modified  when strided tensors are handled
+  std::string expected = "for (int c2 = t1; c2 <= 31; c2 += 8)";
```

Comment:
this does not check strides, just that some loop is set
you want to look at the place where the case from pointer to multidimensional array occurs and create an assertion based on that.
At this time the cast will only take sizes into account and we will need it to take strides into account that do not yet exist.

Reply:
Thanks! I see your point about asserting based on the cast to multi-dim array. I'll make that change.